recipe-client-addon/lib/PreferenceExperiments.jsm

```diff
@@ -0,0 +1,296 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Preference Experiments temporarily change a preference to one of several test
+ * values for the duration of the experiment. Telemetry packets are annotated to
+ * show what experiments are active, and we use this data to measure the
+ * effectiveness of the preference change.
+ *
+ * Info on active and past experiments is stored in a JSON file in the profile
+ * folder.
+ *
+ * Active preference experiments are stopped if they aren't active on the recipe
+ * server. They also expire if Firefox isn't able to contact the recipe server
+ * after a period of time, as well as if the user modifies the preference during
+ * an active experiment.
+ */
+
+/**
+ * Experiments store info about an active or expired preference experiment.
+ * They are single-depth objects to simplify cloning.
+ * @typedef {Object} Experiment
+ * @property {string} name
+ *   Unique name of the experiment
+ * @property {string} branch
+ *   Experiment branch that the user was matched to
+ * @property {boolean} expired
+ *   If false, the experiment is active.
+ * @property {string} lastSeen
+ *   ISO-formatted date string of when the experiment was last seen from the
```

Comment:
Because it's what `Date.prototype.toJSON` returns, and these objects get persisted to a JSON file.

Right now we simply update the date, so storing it as a string isn't inconvenient, but a future PR will be comparing it to the current date, which would require converting it. I figured that converting it to a date at that one point wouldn't be an issue. 

Is there an established preference for storing dates in Firefox?